pkg/utils/secretsUtils.go

```diff
@@ -29,14 +30,40 @@ func (clientSets ClientSets) ValidateSecrets(secretName string, experiment *Expe
 }
 
 // SetSecrets sets the value of secrets in Experiment Structure
-func (expDetails *ExperimentDetails) SetSecrets(clients ClientSets) {
+func (expDetails *ExperimentDetails) SetSecrets(clients ClientSets, engineName string) {
 
 	chaosExperimentObj, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(expDetails.Namespace).Get(expDetails.Name, metav1.GetOptions{})
 	if err != nil {
-		log.Infof("Unable to get ChaosEXperiment Resource, wouldn't not be able to patch ConfigMaps")
+		log.Infof("Unable to get ChaosEXperiment Resource, wouldn't be able to patch Secrets")
+	}
+	experimentSecrets := chaosExperimentObj.Spec.Definition.Secrets
+
+	chaosEngineObj, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosEngines(expDetails.Namespace).Get(engineName, metav1.GetOptions{})
+	if err != nil {
+		log.Infof("Unable to patch Secrets, error: %v", err)
 	}
-	secrets := chaosExperimentObj.Spec.Definition.Secrets
-	expDetails.Secrets = secrets
+	experimentsList := chaosEngineObj.Spec.Experiments
+	for i := range experimentsList {
+		if experimentsList[i].Name == expDetails.Name {
+			engineSecrets := experimentsList[i].Spec.Components.Secrets
+			for j := range engineSecrets {
+				flag := false
+				for k := range experimentSecrets {
+					if engineSecrets[j].Name == experimentSecrets[k].Name {
```

Comment:
Please write separate util for this merge, and add comments, or make the code more readable.

Reply:
See if you can do this also in this PR, causing tiny effect on BCH.

Reply:
done